src/App.tsx

```diff
@@ -34,62 +100,131 @@ export function getReorderedGoods(
   // To avoid the original array mutation
   const visibleGoods = [...goods];
 
+  visibleGoods.sort((first, second) => {
```

Comment:
You could sort `visible goods` only if sortType doesn't equal `SortType.None`.
In that case, we could avoid redundant computations